conda/core/index.py

```diff
@@ -51,7 +58,457 @@ def check_allowlist(channel_urls: list[str]) -> None:
+        self.track_features = context.track_features
+        self.add_system = add_system
+
+    def _make_virtual_package(
+        self, name: str, version: str | None = None, build_string: str | None = None
+    ) -> PackageRecord:
+        """
+        Create a virtual package record.
+
+        :param name: The name of the virtual package.
+        :param version: The version of the virtual package, defaults to "0".
+        :param build_string: The build string of the virtual package, defaults to "0".
+        :return: A PackageRecord representing the virtual package.
+        """
+        return PackageRecord(
+            package_type=PackageType.VIRTUAL_SYSTEM,
+            name=name,
+            version=version or "0",
+            build_string=build_string or "0",
+            channel="@",
+            subdir=context.subdir,
+            md5="12345678901234567890123456789012",
+            build_number=0,
+            fn=name,
+        )
+
+    @property
+    def system_packages(self):
+        try:
+            return self._system_packages
+        except AttributeError:
+            self._system_packages = {
+                (
+                    rec := self._make_virtual_package(
+                        f"__{package.name}", package.version, package.build
+                    )
+                ): rec
+                for package in context.plugin_manager.get_virtual_packages()
+            }
+        return self._system_packages
+
+    @property
+    def features(self):
+        try:
+            return self._features
+        except AttributeError:
+            self._features = {
+                (rec := make_feature_record(feature)): rec
+                for feature in context.track_features
+            }
+        return self._features
+
+    @property
+    def prefix_data(self):
+        if self._prefix_data is None and self.prefix_path:
+            self._prefix_data = PrefixData(self.prefix_path)
+        return self._prefix_data
+
+    def reload(self, prefix=False):
+        if prefix:
+            if self.prefix_data:
+                self.prefix_data.reload()
+
+    def __repr__(self):
+        channels = ", ".join(self.channels.keys())
+        return f"Index(channels=[{channels}])"
+
+    def get_reduced_index(self, specs):
+        return ReducedIndex(
+            specs=specs,
+            channels=self._channels,
+            prepend=False,
+            subdirs=self._subdirs,
+            use_local=False,
+            use_cache=self.use_cache,
+            prefix=self.prefix_path,
+            repodata_fn=self._repodata_fn,
+            add_system=self.add_system,
+        )
+
+    @property
+    def data(self):
+        try:
+            return self._data
+        except AttributeError:
+            self._realize()
+            return self._data
+
+    @data.setter
+    def data(self, value):
+        self._data = value
+
+    def _supplement_index_dict_with_prefix(self, index_dict):
```

Comment:
Do we need this as an instance method?

```suggestion
    @staticmethod
    def supplement_index_dict_with_prefix(index_dict):
```